### DISCUSSION TITLE
The workaround in digest_dupctx does not clear the error raised in p11prov_GetOperationState

### DISCUSSION THREAD

**Discussion by 0pq76r** [openssl-projects/pkcs11-provider#320] [General]
When the token returns an error when calling C_GetOperationState in the p11prov_digest_dupctx function, the current policy is to continue and returning the "moved" context instead of a copy.

The problem I'm facing with that approach is that when calling p11prov_GetOperationState it calls P11PROV_raise. If the caller decides to check for SSL errors (which nginx does), it will abort the operation despite everything running successfully. Though annoying, I think that is the correct behaviour of the caller.

Would it be possible to add a quirk to suppress/clear that error or to skip calling C_GetOperationState in the first place?

**Comment by simo5** [comment]
I think we should have a quirk to skip calling C_GetOperationState completely, most tokens do not support it.
